cargo-near/src/build.rs

```diff
@@ -1,19 +1,23 @@
 use crate::abi::{AbiCompression, AbiFormat, AbiResult};
 use crate::cargo::{manifest::CargoManifestPath, metadata::CrateMetadata};
+use crate::util::StepPrinter;
 use crate::{abi, util, BuildCommand};
 use colored::Colorize;
 use std::io::BufRead;
 
 const COMPILATION_TARGET: &str = "wasm32-unknown-unknown";
 
 pub(crate) fn run(args: BuildCommand) -> anyhow::Result<()> {
+    let mut step_printer = StepPrinter::new(7);
```

Comment:
if `--no-abi` is activated, won't this affect the count?

Reply:
You are right. Hmm, I thought having statically known build step count could be cool for progress visibility, but now I am thinking it might become a nightmare to support long term. WDYT? Worth bothering with?

Reply:
Yeah, it'll be quite problematic. What do you think about this pattern:

```console
 [•] Checking Environment... [done]
 [•] Collecting cargo project metadata... [done]
 [•] Extracting ABI from contract
    ... cargo stderr ...
 [•] Compressing ABI to be embedded... [done]
 [•] Building Contract (with embedded ABI)
    ... cargo stderr ...
 [✓] Contract Successfully Built!
```

would be neat if we could also indent cargo `std{out,err}` output to follow the style.

Reply:
Played with the colors a bit and this is what I have so far. LMK what you think:

![latest-screenshot](https://user-images.githubusercontent.com/7879134/195802662-8d8d0521-9228-4835-a103-b60f5dd6f37f.png)

Indenting the cargo output turned out to be a little troublesome, but I think I have figured it out in the end. A bit annoying that they are aligning the step names by the right side, but oh well.

Reply:
> A bit annoying that they are aligning the step names by the right side, but oh well.

Is there a chance we can align it to match cargo's? how would that even look.